lib/db_utils.go

```diff
@@ -1184,14 +1192,12 @@ func DBGetWithTxn(txn *badger.Txn, snap *Snapshot, key []byte) ([]byte, error) {
 		return nil, err
 	}
 
-	// If a flush takes place, we don't update cache. It will be updated in DBSetWithTxn.
+	// TODO: Do we want to update the database cache when performing GETs? I think it would be
+	// safer to ONLY update the cache when performing SETs. This way, we can avoid the possibility
+	// of the cache getting out of sync with the database when a badger view transaction is started
+	// before a badger update transaction begins.
```

Comment:
This seems OK to me because we only get to this line if the item is NOT in the cache yet, which means we've never read it from the DB. If you look above on line 1179 you'll see it FIRST tries to look into the cache, and only gets here if it's NOT in the cache. If you think there's still an issue lmk.

Reply:
I'm just not 100% sure one way or the other. I guess there's maybe a race condition that the value would be flushed while we're doing the get.